### DISCUSSION TITLE
Set global mesh size in Icepak AEDT

### DISCUSSION THREAD

**Discussion by mjmashayekhi** [ansys/pyaedt#4642] [Q&A] [answered]
I'd like to set the global mesh size for my Icepak AEDT simulation in my script. I used the following command for this purpose:

ipk.mesh["GlobalMeshSettings"]["XSize"] = "1mm"
ipk.mesh["GlobalMeshSettings"]["YSize"] = "1mm"
ipk.mesh["GlobalMeshSettings"]["ZSize"] = "0.5mm"

However I get an error message. I'd appreciate your help.

**Comment by lorenzovecchietti** [comment, marked as answer]
Hi, mesh region code has been recently refactored. Most of previous command should work, but not this specific one.

The general approach is to get the mesh region object (in case of the global one: `mesh_region = ipk.mesh.global_mesh_region`) and access the setting property.

You can see the available propertied printing `mesh_region.settings` and modify them accessing the object like a dictionary. To switch from automatic to manual settings: `mesh_region.manual_settings = True` and the available properties in `mesh_region.settings` will automatically update accordingly.

When you are done modifing the settings:
`mesh_region.update()`


In your case:
```
ipk.mesh.global_mesh_region.manual_settings=True
ipk.mesh.global_mesh_region.settings["MaxElementSizeX"]="1mm"
ipk.mesh.global_mesh_region.settings["MaxElementSizeY"]="1mm"
ipk.mesh.global_mesh_region.settings["MaxElementSizeZ"]="0.5mm"
ipk.mesh.global_mesh_region.update()
```

**Reply by mchoi327** [reply]
Hi, does this still work on PyAEDT 0.9.6?

I recently updated PyAEDT from 0.9.3 to 0.9.6, but it does not work at all now, which worked well before.

The code I used is as follows:

ipk.mesh.global_mesh_region.UserSpecifiedSettings = True

ipk.mesh.global_mesh_region.MaxElementSizeX = "500um"
ipk.mesh.global_mesh_region.MaxElementSizeY = "500um"
ipk.mesh.global_mesh_region.MaxElementSizeZ = "50um"

ipk.mesh.global_mesh_region.MinGapX = "2um"
ipk.mesh.global_mesh_region.MinGapY = "2um"
ipk.mesh.global_mesh_region.MinGapZ = "1um"

ipk.mesh.global_mesh_region.MinElementsInGap = 5
ipk.mesh.global_mesh_region.MinElementsOnEdge = 5
ipk.mesh.global_mesh_region.MaxSizeRatio = 2

ipk.mesh.global_mesh_region.NoOGrids = True
ipk.mesh.global_mesh_region.StairStepMeshing = True

ipk.mesh.global_mesh_region.EnableMLM = False

ipk.mesh.global_mesh_region.UniformMeshParametersType = "None"
ipk.mesh.global_mesh_region.MaxLevels = 2
ipk.mesh.global_mesh_region.BufferLayers = 1
ipk.mesh.global_mesh_region.update()

**Reply by lorenzovecchietti** [reply]
Hi, 

as you see in the previous message, the new suggested way to modify settings is via the settings dictionary. Hence, something like:
`ipk.mesh.global_mesh_region.settings["MaxElementSizeZ"]="0.5mm"`

The way you are accessing those settings is partially supported for backward compatibility, but it will be removed in future versions.

I know that this is not reflected on the examples right now, sorry about this. New examples are ready to be published, we are waiting for an internal approval.

In conclusion:

1. Can you try to change the code accordingly and get back to me if something still does not work?
2. If you cannot, change the code, can you share with me what is the error message and can you try to identify what is the settings that cause the error?

**Reply by mchoi327** [reply]
Hi,
Thank you for your prompt response.

Both of the codes you suggested and I mentioned in the reply did not work.

The problem is that both of them return 'True' after taking a lot of time (which was instantaneously reflected on v.0.9.3) and make no changes to the Global Mesh Region, as shown in the figure attached. 

![image](https://github.com/ansys/pyaedt/assets/122655322/c8468bd3-67db-4fc3-a82a-b77d1dd9442f)

I have no control over meshing through PyAEDT at this time.. Would the examples for meshing be published soon?


Also, I want to let you know that 'assign_mesh_region' for non-box-shaped objects is not working with the error message that only box-shaped objects are available for 'assign_mesh_region.'

However, when we set it up manually in GUI, the assigned mesh region worked as Ansys AEDT automatically setting up the bounding box.

**Reply by lorenzovecchietti** [reply]
- Just tested again and actually found a bug. Pushing the fix tomorrow. Thanks for pointing this out!
- Unfortunately, I don't know when the new examples will be released.
- The 'assign_mesh_region' method should work (at least on my side does), but it was changed specifically to match with the new subregion paradigm that was introduced in 24R1. I *guess* that you are using an older version. Unfortunately, maintaining backward compatibility with this "big" changes is hard. In this case I chose not to clutter the code as the workaround should be relatively straightforward (create a region box and then pass it as argument)

**Reply by mchoi327** [reply]
Thanks, Lorenzo for the prompt response.

Do I need to update PyAEDT tomorrow to use the bug-fixed one tomorrow?

Regarding the third bullet, yes, I am using 23R1. I may try with 24R1 in the future. :)
